### PR TITLE
Add setting to disable extensions

### LINKED ISSUE #62: layer: List of extensions to modify extensions on top of the selected profile?
Potential new feature to the Profiles layer: adding a setting with a list of extensions that can be added and enabled.
This list of enabled/disabled extensions would override the extensions on top of the selected profile.

### PR DESCRIPTION
**Pull request by ziga-lunarg**
Closes #62 

TODO:
- Implement the exlcude_formats settings
- Populate the exclude_extensions and exclude_formats with valid values.

### CODE REVIEW

File: layer/VkLayer_khronos_profiles.json.in [hunk 1]
```diff
@@ -170,6 +170,14 @@
                     ],
                     "default": [ "SIMULATE_API_VERSION_BIT", "SIMULATE_FEATURES_BIT", "SIMULATE_PROPERTIES_BIT" ]
                 },
+                {
+                    "key": "disable_extensions",
+                    "label": "Disable extensions",
+                    "description": "Removes the extensions from being reported by the Vulkan physical device.",
+                    "status": "STABLE",
+                    "type": "EXTENSIONS",
+                    "format": "EXTENSION"
```
Comment (ziga-lunarg) [inline layer/VkLayer_khronos_profiles.json.in:179]: @christophe-lunarg I am not sure about this, should this be something else?

File: layer/profiles.cpp [hunk 2]
```diff
@@ -8597,6 +8633,17 @@ VKAPI_ATTR void VKAPI_CALL GetPhysicalDeviceFormatProperties(VkPhysicalDevice ph
                                                              VkFormatProperties *pFormatProperties) {
     LogMessage(DEBUG_REPORT_DEBUG_BIT, "GetPhysicalDeviceFormatProperties\n");
 
+    // Check if Format was excluded
+    for (std::size_t j = 0, m = layer_settings.exclude_formats.size(); j < m; ++j) {
+        const std::string &excluded_format = layer_settings.exclude_formats[j].first;
+        if (excluded_format.empty()) continue;
+
+        if (StringToFormat(excluded_format) == format) {
+            *pFormatProperties = VkFormatProperties{};
```
Comment (christophe-lunarg) [inline layer/profiles.cpp:8642]: Isn't this only partially disable the format properties? What about VkFormatProperties3?

**Comment by christophe-lunarg** [comment]
The PR is also missing the doc update.

File: layer/profiles.cpp (hunk 2 above)
Comment (ziga-lunarg) [reply layer/profiles.cpp:8642]: `GetPhysicalDeviceFormatProperties2` and `GetPhysicalDeviceFormatProperties2KHR` both call this function, so every path with go through this.

**Review by mark-lunarg — APPROVED** [review]
I'll update my PR for the new settings soon.

**Pull request merged**